_extras/extra_challenges.md

```diff
@@ -8,12 +8,31 @@ permalink: /extra_challenges/
 
 A collection of challenges that have been either removed from or not (yet) added to the main lesson. 
 
-## 06-loops-and-functions
-
-### Looping Over Dataframe Challenges
-
-* What happens if there is no data for a year in the sequence (for example, imagine we had used 1976 as the start year in `range`)
-~~~
-An empty file with only the headers
-~~~
-{: .output}
+> ## Looping Over Dataframe
+>
+> (Please refer to lesson `06-loops-and-functions.md`)
+>
+> The file `surveys.csv` in the `data` folder contains 25 years of data from surveys,
+> starting from 1977. We can load the data and print all the years surveyed using a `for` loop:
+> 
+> ~~~
+> import pandas as pd
+> 
+> # Load the data into a DataFrame
+> surveys_df = pd.read_csv('data/surveys.csv')
+> 
+> # Loop through a sequence of years and print the year
+> start_year = 1977
+> end_year = 2002
+> for year in range(start_year, end_year+1):
+>     print(str(year))
```

Comment:
Right, this is a function of the fact that where this challenge was embedded is right before we discuss creating new name strings to automate naming files. It can stay or go, in my opinion. Divorced from the context, perhaps it ought to go.